test/functional/p2p_filter.py

```diff
@@ -125,8 +153,11 @@ def run_test(self):
 
         self.log.info('Check that we receive a tx in reply to a mempool msg if the filter matches a mempool tx')
         filter_node.merkleblock_received = False
+        filter_node.tx_received = False
```

Comment:
Good point. I'm thinking a getter and a setter: `get_tx_received`, `reset_tx_received` (to False), and for `merkleblock_received` as well? Shall I change the rest of the test too since it uses these throughout?

Reply:
> Shall I change the rest of the test too since it uses these throughout?

If you want to, that'd be great! There are lots of data races in this test where the test logic thread and the p2p handler thread can be accessing the same memory.

Take a look at Python properties: https://docs.python.org/3/library/functions.html#property. By implementing a getter/setter using properties, you don't need to change the client code from `assert not filter_node.tx_received` to `assert not filter_node.get_tx_received()`.

Be careful not to use the setter for the `on_xxxx()`, methods, which already have the lock when called. It's better to make the locks non-reentrant if possible (see https://github.com/bitcoin/bitcoin/pull/19178)